### PR TITLE
wait for supported resource types to be ready before starting galley

### PR DESCRIPTION
**Pull request by ayj**
fixes https://github.com/istio/istio/issues/9491

### CODE REVIEW

File: galley/pkg/kube/source/source.go [hunk 1]
```diff
@@ -49,6 +53,68 @@ func New(k kube.Interfaces, resyncPeriod time.Duration, cfg *converter.Config) (
...
+			if err != nil {
+				errs = multierror.Append(errs, fmt.Errorf("could not find %v: %v", name, err))
+				continue nextCRD
+			}
+
+			for _, cond := range crd.Status.Conditions {
+				switch cond.Type {
+				case apiextensionsv1beta1.Established:
+					if cond.Status == apiextensionsv1beta1.ConditionTrue {
+						log.Infof("Found CRD %q", name)
```
Comment (ozevren) [inline galley/pkg/kube/source/source.go]: Please use the right scope.

File: galley/pkg/kube/source/source.go [hunk 2]
```diff
@@ -49,6 +53,68 @@ func New(k kube.Interfaces, resyncPeriod time.Duration, cfg *converter.Config) (
...
+						delete(crdToFind, name)
```
Comment (ozevren) [inline galley/pkg/kube/source/source.go]: Presumably deleting while looping over a slice is safe?

File: galley/pkg/kube/source/source.go [hunk 3]
```diff
@@ -49,6 +53,68 @@ func New(k kube.Interfaces, resyncPeriod time.Duration, cfg *converter.Config) (
 	return newSource(k, resyncPeriod, cfg, kube_meta.Types.All())
 }
 
+// VerifyCRDPresence verifies that all expected CRDs are registered
+// with the k8s kube-apiserver.
+func VerifyCRDPresence(k kube.Interfaces) error {
```
Comment (ozevren) [inline galley/pkg/kube/source/source.go]: Nit: can you place this in a separate file? It is technically not part of the runtime/source model.

File: galley/pkg/kube/source/source.go (hunk 2 above)
Comment (ayj) [reply galley/pkg/kube/source/source.go]: Yes. 

(from https://golang.org/ref/spec#For_statements)
> The iteration order over maps is not specified and is not guaranteed to be the same from one iteration to the next. ___If a map entry that has not yet been reached is removed during iteration, the corresponding iteration value will not be produced___. If a map entry is created during iteration, that entry may be produced during the iteration or may be skipped. The choice may vary for each entry created and from one iteration to the next. If the map is nil, the number of iterations is 0.

**Comment by ayj** [comment]
cc @hklai for Godep.lock update

**Comment by hklai** [comment]
/approve

Gopkg.lock

**Pull request merged**